src/XCCDF/xccdf_session.c

```diff
@@ -1554,7 +1554,11 @@ int xccdf_session_remediate(struct xccdf_session *session)
 	xccdf_result_set_version(session->xccdf.result,
 			benchmark != NULL ? xccdf_benchmark_get_version(benchmark) : NULL);
 	xccdf_result_fill_sysinfo(session->xccdf.result);
-	return xccdf_policy_remediate(xccdf_session_get_xccdf_policy(session), session->xccdf.result);
+
+	if ((res = xccdf_policy_remediate(xccdf_session_get_xccdf_policy(session), session->xccdf.result)) != 0)
```

Comment:
I did not want to change the signature of any of the functions. I added the `if` statement to ensure that in case `xccdf_policy_remediate` gets updated in the future to return anything else, `xccdf_session_remediate` would also return it as before. If the plan is to make it return void, I will remove the `if` statement.